src/Bridge/Laravel/Validation/ExtendedUrlRule.php

```diff
@@ -0,0 +1,68 @@
+declare(strict_types=1);
+
+namespace EoneoPay\Externals\Bridge\Laravel\Validation;
+
+use Closure;
+use EoneoPay\Externals\Bridge\Laravel\Interfaces\ValidationRuleInterface;
+
+/**
+ * This rule is almost the same as Laravel's "url" rule, but allows urls without protocol, e.g. "//example.com".
+ */
+final class ExtendedUrlRule implements ValidationRuleInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getName(): string
+    {
+        return 'extended_url';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getReplacements(): Closure
+    {
+        return static function (string $message, string $attribute): string {
+            return \str_replace(':attribute', $attribute, $message);
+        };
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
+     * @noinspection PhpUnusedParameterInspection
+     */
+    public function getRule(): Closure
+    {
+        return static function (string $attribute, $value): bool {
+            if (\is_string($value) === false) {
+                return false;
+            }
+
+            /** @noinspection NotOptimalRegularExpressionsInspection */
+            /** @codingStandardsIgnoreStart */
+            $pattern = '~^
+            (((aaa|aaas|about|acap|acct|acr|adiumxtra|afp|afs|aim|apt|attachment|aw|barion|beshare|bitcoin|blob|bolo|callto|cap|chrome|chrome-extension|cid|coap|coaps|com-eventbrite-attendee|content|crid|cvs|data|dav|dict|dlna-playcontainer|dlna-playsingle|dns|dntp|dtn|dvb|ed2k|example|facetime|fax|feed|feedready|file|filesystem|finger|fish|ftp|geo|gg|git|gizmoproject|go|gopher|gtalk|h323|ham|hcp|http|https|iax|icap|icon|im|imap|info|iotdisco|ipn|ipp|ipps|irc|irc6|ircs|iris|iris.beep|iris.lwz|iris.xpc|iris.xpcs|itms|jabber|jar|jms|keyparc|lastfm|ldap|ldaps|magnet|mailserver|mailto|maps|market|message|mid|mms|modem|ms-help|ms-settings|ms-settings-airplanemode|ms-settings-bluetooth|ms-settings-camera|ms-settings-cellular|ms-settings-cloudstorage|ms-settings-emailandaccounts|ms-settings-language|ms-settings-location|ms-settings-lock|ms-settings-nfctransactions|ms-settings-notifications|ms-settings-power|ms-settings-privacy|ms-settings-proximity|ms-settings-screenrotation|ms-settings-wifi|ms-settings-workplace|msnim|msrp|msrps|mtqp|mumble|mupdate|mvn|news|nfs|ni|nih|nntp|notes|oid|opaquelocktoken|pack|palm|paparazzi|pkcs11|platform|pop|pres|prospero|proxy|psyc|query|redis|rediss|reload|res|resource|rmi|rsync|rtmfp|rtmp|rtsp|rtsps|rtspu|secondlife|s3|service|session|sftp|sgn|shttp|sieve|sip|sips|skype|smb|sms|smtp|snews|snmp|soap.beep|soap.beeps|soldat|spotify|ssh|steam|stun|stuns|submit|svn|tag|teamspeak|tel|teliaeid|telnet|tftp|things|thismessage|tip|tn3270|turn|turns|tv|udp|unreal|urn|ut2004|vemmi|ventrilo|videotex|view-source|wais|webcal|ws|wss|wtai|wyciwyg|xcon|xcon-userid|xfire|xmlrpc\.beep|xmlrpc.beeps|xmpp|xri|ymsgr|z39\.50|z39\.50r|z39\.50s)):)?//                                 # protocol
```

Comment:
I dont agree with the protocol options being listed like this: there can be all kinds of custom protocols and we should be using a simpler constraint:

https://github.com/symfony/validator/blob/master/Constraints/UrlValidator.php

Reply:
In your example pattern will eventually contain only http/https after sprintf. Do you want me to put only http/https there? Or allow any alphanum string (with dashes and dots)?

Reply:
@merk